files/bootstrap.sh

```diff
@@ -135,18 +135,56 @@ set -- "${POSITIONAL[@]}" # restore positional parameters
 CLUSTER_NAME="$1"
 set -u
 
+KUBELET_VERSION=$(kubelet --version | grep -Eo '[0-9]\.[0-9]+\.[0-9]+')
+echo "Using kubelet version $KUBELET_VERSION"
+
+function is_greater_than_or_equal_to_version() {
+    local actual_version="$1"
+    local compared_version="$2"
+
+    [ $actual_version = "`echo -e \"$actual_version\n$compared_version\" | sort -V | tail -n1`" ]
+}
+
+# As of Kubernetes version 1.24, we will start defaulting the container runtime to containerd
+# and no longer support docker as a container runtime.
+IS_124_OR_GREATER=false
+DEFAULT_CONTAINER_RUNTIME=dockerd
+if is_greater_than_or_equal_to_version $KUBELET_VERSION "1.24.0"; then
+    IS_124_OR_GREATER=true
+    DEFAULT_CONTAINER_RUNTIME=containerd
+fi
+
+# Set container runtime related variables
+DOCKER_CONFIG_JSON="${DOCKER_CONFIG_JSON:-}"
+ENABLE_DOCKER_BRIDGE="${ENABLE_DOCKER_BRIDGE:-false}"
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-$DEFAULT_CONTAINER_RUNTIME}"
+
+echo "Using $CONTAINER_RUNTIME as the container runtime"
+
+if $IS_124_OR_GREATER; then
+    if [ $CONTAINER_RUNTIME != "containerd" ]; then
```

Comment:
Is there any reason not to use some of the existing conditions?

https://github.com/awslabs/amazon-eks-ami/blob/3a7d9744f11652cada6a1a29b878a9e3e21dbb37/files/bootstrap.sh#L604-L626

Reply:
That's fair. The containerd warning logs are pertinent any time you're setting them with containerd as the runtime.